### PR TITLE
Adding features and improvements.

### PR DESCRIPTION
**Pull request by Gerrit91**
- Add `ip allocate` command to directly allocate an IP address (also specific IP if desired)
- Makes `name` and `description` required fields on `ip static` command
- Clears the CCM IP tags on `ip static` because it does not make sense on a static IP address anymore
- Adds warnings for `ip static` and `ip allocate`
- Add global flag `--yes-i-really-mean-it` to skip security prompts
- Alternative reconcile operations `--retry` and `--maintain`

### CODE REVIEW

File: cmd/root.go [hunk 1]
```diff
@@ -75,6 +75,7 @@ func init() {
 	cloudctl cluster ls -o template --template "{{ .metadata.uid }}"
 
 	`)
+	rootCmd.PersistentFlags().BoolP("yes-i-really-mean-it", "", false, "skips security prompts (which can be dangerous to set blindly because actions can lead to data loss or additional costs)")
```
Comment (majst01) [inline cmd/root.go:78]: please align with:
viper.GetBool("i-am-aware-of-dangerous-settings")

Comment (Gerrit91) [reply cmd/root.go:78]: Done.

**Review by majst01 — APPROVED** [review]

File: go.mod [hunk 2]
```diff
@@ -5,7 +5,7 @@ go 1.15
 require (
 	github.com/Masterminds/semver v1.5.0
 	github.com/fatih/color v1.9.0
-	github.com/fi-ts/cloud-go v0.8.0
+	github.com/fi-ts/cloud-go v0.8.1
 	github.com/gardener/gardener v1.8.2
 	github.com/go-openapi/runtime v0.19.21
```
Comment (majst01) [inline go.mod:10]: can you please update metal-lib to v0.6.0 an security to v0.4.0 as well

Comment (Gerrit91) [reply go.mod:10]: Done.

**Pull request merged**